swap/simulations_test.go

```diff
@@ -0,0 +1,704 @@
+// which contains the shared objects needed to initialize the `SimulatedBackend`
+func newSimServiceMap(params *swapSimulationParams) map[string]simulation.ServiceFunc {
+	simServiceMap := map[string]simulation.ServiceFunc{
+		// we need the bzz service in order to build up a kademlia
+		"bzz": func(ctx *adapters.ServiceContext, bucket *sync.Map) (node.Service, func(), error) {
+			addr := network.NewAddr(ctx.Config.Node())
+			hp := network.NewHiveParams()
+			hp.Discovery = false
+			config := &network.BzzConfig{
+				OverlayAddr:  addr.Over(),
+				UnderlayAddr: addr.Under(),
+				HiveParams:   hp,
+			}
+			kad := network.NewKademlia(addr.Over(), network.NewKadParams())
+			return network.NewBzz(config, kad, nil, nil, nil), nil, nil
+		},
+		// and we also use a swap service
+		"swap": func(ctx *adapters.ServiceContext, bucket *sync.Map) (s node.Service, cleanup func(), err error) {
+			// every simulation node has an instance of a `testService`
+			ts := newTestService()
+			// balance is the interface for `NewAccounting`; it is a Swap
+			balance := params.swaps[params.count]
+			// every node is a different instance of a Swap and gets a different entry in the map
+			params.count++
+			// to create the accounting, we also need a `Price` instance
+			prices := &testPrices{}
+			// create the matrix of test prices
+			prices.newTestPriceMatrix()
+			ts.spec = newTestSpec()
+			// create the accounting instance and assign to the spec
+			ts.spec.Hook = protocols.NewAccounting(balance, prices)
+			ts.swap = balance
+			// deploy the accounting to the `SimulatedBackend`
+			testDeploy(context.Background(), balance.backend, balance)
+			params.backend.Commit()
+			// store the testService into the bucket
+			bucket.Store(bucketKeySwap, ts)
+
+			cleanup = func() {
+				for _, dir := range params.dirs {
+					os.RemoveAll(dir)
+				}
+			}
+
+			return ts, cleanup, nil
+		},
+	}
+	return simServiceMap
+}
+
+// newSharedBackendSwaps pre-loads each simulated node account with "funds"
+// so that later in the simulation all operations have sufficient gas
+func newSharedBackendSwaps(nodeCount int) (*swapSimulationParams, error) {
+	params := &swapSimulationParams{
+		swaps:       make(map[int]*Swap),
+		dirs:        make(map[int]string),
+		maxMsgPrice: 10000,
+		minMsgPrice: 100,
+		nodeCount:   nodeCount,
+	}
+	keys := make(map[int]*ecdsa.PrivateKey)
+	addrs := make(map[int]common.Address)
+	alloc := core.GenesisAlloc{}
+	stores := make(map[int]*state.DBStore)
+
+	// for each node, generate keys, a GenesisAccount and a state store
+	for i := 0; i < nodeCount; i++ {
+		key, err := crypto.GenerateKey()
+		if err != nil {
+			return nil, err
+		}
+		keys[i] = key
+		addrs[i] = crypto.PubkeyToAddress(key.PublicKey)
+		alloc[addrs[i]] = core.GenesisAccount{Balance: big.NewInt(10000000000)}
+		dir, err := ioutil.TempDir("", fmt.Sprintf("swap_test_store_%d", i))
+		if err != nil {
+			return nil, err
+		}
+		stateStore, err2 := state.NewDBStore(dir)
```

Comment:
Not sure where exactly but it seems there are some file handles which are not properly cleaned up. Running tests with a higher `-count` always leads to `too many open files` errors.

Reply:
Not sure if this was related to the ulimit issue on my side or not (this error also happened on the other terminal I believe) but I don't get the error anymore on high counts so I assume this is solved.